src/typeahead/highlight.ts

```diff
@@ -0,0 +1,35 @@
+import {Component, Input, OnChanges, ChangeDetectionStrategy} from '@angular/core';
+import {regExpEscape} from '../util/util';
+
+@Component({
+  selector: 'ngb-highlight',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<template ngFor [ngForOf]="parts" let-part let-isOdd="odd">` +
+      `<span *ngIf="isOdd" class="{{highlightClass}}">{{part}}</span><template [ngIf]="!isOdd">{{part}}</template>` +
+      `</template>`,  // template needs to be formatted in a certain way so we don't add empty text nodes
+  styles: [`
+    .ngb-highlight {
+      font-weight: bold;
+    }
+  `]
+})
+export class NgbHighlight implements OnChanges {
+  private parts: string[];
+
+  @Input() highlightClass = 'ngb-highlight';
+  @Input() result: string;
```

Comment:
`result` seems strange to me here as a name if you look at it as a standalone component:

```html
<ngb-highlight result="something to highlight" term="term">
```

maybe rename to something like `input`?